cli/command/cli.go

```diff
@@ -85,6 +87,9 @@ type DockerCli struct {
 	dockerEndpoint     docker.Endpoint
 	contextStoreConfig store.Config
 	initTimeout        time.Duration
+	res                *resource.Resource
```

Comment:
nit: From what I see, I think we could group these 3 into a single struct defined in the `telemetry` file, so it's not too tightly coupled with this cli struct and so that it's purpose is a bit clearer (this struct is getting a bit wild imho).

The only part of the telemetry code that seems to need data from the actual DockerCli struct is `dockerExporterOtlpEndpoint` (which only needs the metadata).

Any opinions on this?

Reply:
Yea, this can probably be wrapped into its own struct to make it a bit less unwieldy. I'll work on that to see how well it works.

Reply:
Moved these three attributes into its own struct and modified the infrastructure around it.

I felt like the method implementations of the tracer provider and meter provider were likely still more straightforward when attached to the `DockerCli`, but the resource stuff could clearly be moved to its own separate thing without making things more complicated.